scripts/reindex.py

```diff
@@ -96,8 +108,21 @@ def main(argv: List[str]):
 
     configure_script_logging(logger)
 
+    source_globs = set(args.sources)
+    if source_globs != {'*'} and not args.partition_prefix_length:
+        parser.error('Cannot specify sources when performing a local reindex')
+
     azul = AzulClient(num_workers=args.num_workers)
 
+    sources_by_catalog = defaultdict(set)
+    for catalog in args.catalogs:
+        sources = azul.catalog_sources(catalog)
+        for source_glob in source_globs:
+            matches = fnmatch.filter(sources, source_glob)
+            require(matches, 'No matching source in catalog', source_glob, catalog)
```

Comment:
I think we should allow there to be no matches. If I understand this correctly, I'd be forced to specify the catalog of a source if I specify a source because the default value for the catalogs option would likely reference a catalog that doesn't match the source. That's impractical, I think.